lighthouse-core/gather/driver.js

```diff
@@ -247,14 +234,16 @@ class Driver {
    * user of that domain (e.g. two gatherers have enabled a domain, both need to
    * disable it for it to be disabled). Returns true otherwise.
    * @param {string} domain
+   * @param {string|undefined} sessionId
    * @param {boolean} enable
    * @return {boolean}
    * @private
    */
-  _shouldToggleDomain(domain, enable) {
-    const enabledCount = this._domainEnabledCounts.get(domain) || 0;
+  _shouldToggleDomain(domain, sessionId, enable) {
```

Comment:
time to add tests for this functionality? :D

should hopefully be pretty easy with the new mock capabilities and can just `sendCommand('Network.enable')` or whatever to make sure it doesn't turn off to early :)

Reply:
done